website_lazy_load_image/models/ir_ui_view.py

```diff
@@ -21,7 +21,8 @@ def render_template(self, template, values=None, engine='ir.qweb'):
         res = super(IrUiView, self).render_template(template, values, engine)
         website_id = self.env.context.get('website_id')
         if website_id and not \
-                self.env['website'].browse(website_id).is_publisher():
+                self.env['website'].browse(website_id).is_publisher() and \
+                '<html' in res.decode():
```

Comment:
Is there no better way to check if it's HTML?

Reply:
I don't know, I'm not familiar with Odoo, but I agree it's not pretty.

Reply:
Is _robots.txt_ the only case? If so you can just do a check `if template == 'website.robots'`. I'm concerned that `<html in res.decode()` will make lazy loading on fixed pages like contact or forum not work anymore.

Reply:
I would not add an exception for specific templates...
Anyway what if you get HTML but no `<html />` tag?
In theory, something like this should do trick:
```py
>>> lxml.html.fromstring('no html').find('.//')
>>> lxml.html.fromstring('<html></html>').find('.//')
>>> lxml.html.fromstring('<html><body></body></html>').find('.//')
<Element body at 0x7f61236c5c78>
```
case 1 and 2: do nothing, case 3: go on.
In this case you should replace `etree.HTML(res)` of course.
Sounds good?